src/ark/utils/plot_utils.py

```diff
@@ -170,24 +173,113 @@ def assign_metacluster_cmap(self, fov_img: np.ndarray) -> np.ndarray:
         return relabeled_fov
 
 
-def plot_neighborhood_cluster_result(img_xr, fovs, k, save_dir=None, cmap_name='tab20',
-                                     fov_col='fovs', figsize=(10, 10)):
-    """Takes an xarray containing labeled images and displays them.
+def plot_cluster(
+        image: np.ndarray,
+        fov: str,
+        cmap: colors.ListedColormap,
+        norm: colors.BoundaryNorm,
+        cbar_visible: bool = True,
+        cbar_labels: list[str] = None,
+        dpi: int = 300,
+        figsize: tuple[int, int] = (10, 10)) -> Figure:
+    """
+    Plots the cluster image with the provided colormap.
+
     Args:
-        img_xr (xarray.DataArray):
-            xarray containing labeled cell objects.
-        fovs (list):
-            list of fovs to display.
+        image (np.ndarray):
+            The cluster image to plot.
+        fov (str):
+            The name of the clustered FOV.
+        cmap (colors.ListedColormap):
+            A colormap to use for the cluster image.
+        norm (colors.BoundaryNorm):
+            A normalization to use for the cluster image.
+        cbar_visible (bool, optional):
+            Whether or not to display the colorbar. Defaults to True.
+        cbar_labels (list[str], optional):
+            Colorbar labels for the clusters. Devaults to None, where
+            the labels will be automatically generated.
+        dpi (int, optional):
+            The resolution of the image to use for saving. Defaults to 300.
+        figsize (tuple, optional):
+            The size of the image to display. Defaults to (10, 10).
+
+    Returns:
+        Figure: Returns the cluster image as a matplotlib Figure.
+    """
+    # Default colorbar labels
+    if cbar_labels is None:
+        cbar_labels = [f"Cluster {x}" for x in range(1, len(cmap.colors))]
+
+    fig: Figure = plt.figure(figsize=figsize, dpi=dpi)
+    fig.set_layout_engine(layout="tight")
+    gs = gridspec.GridSpec(nrows=1, ncols=2, figure=fig, width_ratios=[60, 1])
+    fig.suptitle(f"{fov}")
+
+    # Image axis
+    ax: Axes = fig.add_subplot(gs[0, 0])
+    # Colorbar Axis
+    cax: Axes = fig.add_subplot(gs[0, 1])
+
+    ax.axis("off")
+    ax.grid(visible=False)
+
+    ax.imshow(
+        X=image,
+        cmap=cmap,
+        norm=norm,
+        origin="upper",
+        aspect="equal",
+        interpolation=None,
+    )
+    
+    if cbar_visible:
+        # Manually set the colorbar
+        cbar = fig.colorbar(cm.ScalarMappable(norm=norm, cmap=cmap), cax=cax,
+                            orientation="vertical", use_gridspec=True, pad=0.1)
+        cbar.ax.set_yticks(
+            ticks=np.arange(len(cbar_labels)),
+            labels=cbar_labels
+        )
+        cbar.minorticks_off()
+    else:
+        cax.axis("off")
```

Comment:
Does `GridSpec` require the definition of all elements to format correctly? If not, can we wait until the `cbar_visible` case to define `cax`? It's a bit awkward to define it outside this control flow and have the `else` statement simply turn it off.

Reply:
It may be kinda tricky to do this, as the `GridSpec` currently creates a 1 by 2 grid, where one poriton is for the plot, and the other is for the colorbar. These axes get defined at initialization. I'll adjust it to set the colorbar beforehand.